lib/ex_machina.ex

```diff
@@ -96,6 +100,32 @@ defmodule ExMachina do
   end
 
   @doc """
+  Builds a factory with the passed in factory_name and returns its fields
+
+  ## Example
+
+      def factory(:user) do
+        %MyApp.User{name: "John Doe", admin: false}
+      end
+
+      # Returns %{name: "John Doe", admin: true}
+      fields_for(:user, admin: true)
+  """
+  def fields_for(module, factory_name, attrs \\ %{}) do
+    module.build(factory_name, attrs)
+    |> drop_undesireable_fields
+  end
+
+  defp drop_undesireable_fields(record = %{__struct__: struct}) do
+    record
+    |> Map.drop([:__struct__, :__meta__])
+    |> Map.drop(struct.__schema__(:associations))
+  end
+  defp drop_undesireable_fields(record) do
+    record
```

Comment:
I'm leaning toward no. If someone just wants a map, they can use `Map.from_struct`. I think this function is supposed to be used for Ecto records, but I see your point. Maybe we try raising the error and see if we actually need/want fields_for to convert to a struct if it's not an Ecto struct?